test/clientside/independent-overrides.js

```diff
@@ -10,8 +10,8 @@ var foober =  proxyquire('../fixtures/foo', { './bar': barber });
 var foo    =  proxyquire('../fixtures/foo', { './bar': { } });
 
 test('\noverriding bar.bar for foober but not for foo', function (t) {
-  t.equal(window.foostats.fooRequires(), 3, 'foo is required three times since one for each test and one for require detective')
```

Comment:
Ok, but this test had to change due to this behavior change right?
So it wasn't wrong (not wrong now either), but having to change it reflects that we broke backwards compat.
Some people's tests may depend on the previous behavior.
So we'll need to bump major version for this (or at a minimum minor).